packages/styling/src/styles/getFocusStyle.ts

```diff
@@ -1,31 +1,24 @@
 import { IRawStyle } from '@uifabric/merge-styles';
-import { ITheme } from '../interfaces/index';
+import { IGetFocusStylesOptions, ITheme } from '../interfaces/index';
 import { HighContrastSelector } from './CommonStyles';
 import { IsFocusVisibleClassName } from '@uifabric/utilities';
 import { ZIndexes } from './zIndexes';
 
 /**
  * Generates a focus style which can be used to define an :after focus border.
- *
  * @param theme - The theme object to use.
- * @param inset - The number of pixels to inset the border.
- * @param position - The positioning applied to the container. Must
- * be 'relative' or 'absolute' so that the focus border can live around it.
- * @param highContrastStyle - Style for high contrast mode.
- * @param borderColor - Color of the border.
- * @param outlineColor - Color of the outline.
- * @param isFocusedOnly - If the styles should apply on focus or not.
- * @returns The style object.
+ * @param options - Options to customize the focus border.
  */
-export function getFocusStyle(
-  theme: ITheme,
-  inset: number = 0,
-  position: 'relative' | 'absolute' = 'relative',
-  highContrastStyle: IRawStyle | undefined = undefined,
-  borderColor: string = theme.palette.white,
-  outlineColor: string = theme.palette.neutralSecondary,
-  isFocusedOnly: boolean = true
-): IRawStyle {
+export function getFocusStyles(theme: ITheme, options: IGetFocusStylesOptions = {}): IRawStyle {
```

Comment:
Rather than adding a new method name, you should add a new signature for the existing method. I tried this in my editor and it seems to work:
```ts
/**
 * Generates a focus style which can be used to define an :after focus border.
 *
 * @param theme - The theme object to use.
 * @param options - Options to customize the focus border.
 * @returns The style object.
 */
export function getFocusStyle(theme: ITheme, options?: IGetFocusStyleOptions): IRawStyle;
/**
 * Generates a focus style which can be used to define an :after focus border.
 *
 * @param theme - The theme object to use.
 * @param inset - The number of pixels to inset the border.
 * @param position - The positioning applied to the container. Must
 * be 'relative' or 'absolute' so that the focus border can live around it.
 * @param highContrastStyle - Style for high contrast mode.
 * @param borderColor - Color of the border.
 * @param outlineColor - Color of the outline.
 * @param isFocusedOnly - If the styles should apply on focus or not.
 * @returns The style object.
 * @deprecated Use the object parameter version instead.
 */
export function getFocusStyle(
  theme: ITheme,
  inset: number,
  position: 'relative' | 'absolute',
  highContrastStyle: IRawStyle | undefined,
  borderColor: string,
  outlineColor: string,
  isFocusedOnly: boolean
): IRawStyle;
export function getFocusStyle(
  theme: ITheme,
  insetOrOptions?: number | IGetFocusStyleOptions,
  position?: 'relative' | 'absolute',
  highContrastStyle?: IRawStyle,
  borderColor?: string,
  outlineColor?: string,
  isFocusedOnly?: boolean
): IRawStyle {
  if (typeof insetOrOptions === 'number' || !insetOrOptions) {
    return _getFocusStyleInternal(theme, { inset: insetOrOptions, position, highContrastStyle, borderColor, outlineColor, isFocusedOnly })
  } else {
    return _getFocusStyleInternal(theme, insetOrOptions);
  }
}

function _getFocusStyleInternal(theme: ITheme, options: IGetFocusStyleOptions): IRawStyle {
  const {
    inset = 0,
    position = 'relative',
    highContrastStyle,
    borderColor = theme.palette.white,
    outlineColor = theme.palette.neutralSecondary,
    isFocusedOnly = true
  } = options;

  // rest of code
}
```

Reply:
Actually since this is targeting the Fabric 7 branch, you could entirely remove the old signature. But the solution in my comment above could safely be checked in to Fabric 6 as well if you want.

Reply:
I'm trying that out now, but I'm getting a build error on a line that uses it like this:
```
getFocusStyle(theme, -1, 'relative', cellHighContrastFocus)
```
The error is:  `error TS2575: No overload expects 4 arguments, but overloads do exist that expect either 1 or 7 arguments.`

Reply:
@micahgodbolt Suggested I just deprecate the old function and add a new one for Fabric 7 instead of removing it entirely.

Reply:
Oh oops, all the params except `theme` in the second signature should have ? after them. I'll update the comment.

Reply:
Providing a default for a parameter within the signature makes it automatically optional, but now that I removed the defaults, they had to be explicitly marked as optional with `?`

Reply:
Thanks for pointing out that I can do that, I didn't know it was possible!